File: solutions/python/inputs.py

import splunklib.client as client
import utils
import sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
try: 
    if len(sys.argv) == 1:
        print "syntax: inputs.py [list | show inputname | delete inputname | create inputname index]"
        sys.exit()
    service = client.connect(**opts.kwargs)

    if sys.argv[1] == "list":
        for ip in service.inputs:
            print "  %s" % (ip.name)
    elif sys.argv[1] == "show":
        # need a work-around since service.indexes[path] does not work
        for ip in service.inputs:
            if ip.name == sys.argv[2]:
                for key in sorted(ip.state.keys()):
                    print '  ' + key
                    for sub in ip[key]:
                        print "    %s: %s" % (sub, ip[key][sub]) 
                break
    elif sys.argv[1] == "create":
        ip = service.inputs.create(sys.argv[2], "monitor", **{"index" : sys.argv[3]})
        # raises 404 error even though it works
        print "  created input %s" % ip.name
    elif sys.argv[1] == "delete":
        raise Exception("delete is not currently supported")
        # Indexs.remove() does not work for same reason as given in "list"
    else:
        raise Exception("unknown command \"%s\"" % sys.argv[1])

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: solutions/java/inputs.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class inputs {

    public static void main(String[] args) {

		try {
			if(args.length < 1) {
				System.out.println("syntax: inputs [list | show inputname | create inputname index | delete inputname]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        if(args[0].equalsIgnoreCase("list")) {
				for (String inputName : service.getInputs().keySet())
					System.out.println("  " + inputName);
			}
			else if(args[0].equalsIgnoreCase("show")) {
				Input ip = service.getInputs().get(args[1]);
				for (String key : ip.keySet())
					  System.out.println("  " + key + " = " + ip.get(key));
			}
			else if(args[0].equalsIgnoreCase("create")) {
				Args params = new Args();
				params.put("index", args[2]);
				Input ip = service.getInputs() .create(args[1], InputKind.Monitor, params);
				System.out.println("  input " + ip.getName() + " created for index " + args[2]);
			}
			else if(args[0].equalsIgnoreCase("delete")) {
				service.getInputs().remove(args[1]);
				System.out.println("  " + args[1] + " deleted.");
			}
			else {
				throw new Exception("Unknown command \"" + args[0] + "\"");
			}
			
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}